sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/sqlstore/SQLPlateDAO.java

```diff
@@ -196,6 +197,30 @@ public void setJdbcTemplate(JdbcTemplate template) {
   public void setCascadeType(CascadeType cascadeType) {
     this.cascadeType = cascadeType;
   }
+  
+  public void setAutoGenerateIdentificationBarcodes(boolean autoGenerateIdentificationBarcodes) {
+    this.autoGenerateIdentificationBarcodes = autoGenerateIdentificationBarcodes;
+  }
+  
+  public boolean getAutoGenerateIdentificationBarcodes() {
+    return autoGenerateIdentificationBarcodes;
+  }
+  
+  /**
+   * Generates a unique barcode. Note that the barcode will change when the plate's description is changed.
+   * @param plate
+   */
+  public void autoGenerateIdBarcode(Plate plate) {
```

Comment:
`Plate is a raw type. References to generic type Plate<T,S> should be parameterized`